hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -40,23 +41,32 @@ def load_config(
         if not res.exists():
             raise ConfigLoadError(f"Config not found : {normalized_config_path}")
 
-        with res.open(encoding="utf-8") as f:
-            header_text = f.read(512)
-            header = ConfigSource._get_header_dict(header_text)
-            self._update_package_in_header(
-                header=header,
-                normalized_config_path=normalized_config_path,
-                is_primary_config=is_primary_config,
-                package_override=package_override,
-            )
-            f.seek(0)
-            cfg = OmegaConf.load(f)
-            return ConfigResult(
-                config=self._embed_config(cfg, header["package"]),
-                path=f"{self.scheme()}://{self.path}",
-                provider=self.provider,
-                header=header,
-            )
+        if isinstance(res, zipfile.Path):  # type: ignore
+            with res.open() as f:
+                header_text = f.read(512)
+                header_text = header_text.decode("utf-8")
+                header = ConfigSource._get_header_dict(header_text)
+                f.seek(0)
+                cfg = OmegaConf.load(f)
```

Comment:
I've tested in fbcode with unicode (Chinese) in the config file, OmegaConf can handle it as well, as a result we only need to decode the header text

Reply:
1. looks like zipfile.Path was [introduced in Python 3.8](https://docs.python.org/3/library/zipfile.html). It will be cleaner to test that python is 3.8 or newer instead of checking for the presence of zipfile.Path,
2. Can you minimize the duplicate code between the two branches?
3. I really want to find a way to test this code.